pkg/authenticator/ecrsecret_test.go

```diff
@@ -0,0 +1,222 @@
+package authenticator
+
+import (
+	"context"
+	"encoding/base64"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+)
+
+func TestAuthFilename(t *testing.T) {
+	config := rest.Config{}
+	t.Run("golden path for set HELM_REGISTRY_CONFIG", func(t *testing.T) {
+		testfile := "/test.txt"
+		os.Setenv("HELM_REGISTRY_CONFIG", testfile)
+		ecrAuth, _ := NewECRSecret(&config)
+		val := ecrAuth.AuthFilename()
+
+		assert.Equal(t, val, testfile)
+	})
+
+	t.Run("golden path for no config or secrets", func(t *testing.T) {
+		os.Setenv("HELM_REGISTRY_CONFIG", "")
+		ecrAuth, _ := NewECRSecret(&config)
+		val := ecrAuth.AuthFilename()
+
+		assert.Equal(t, val, "")
+	})
+}
+
+func TestUpdateConfigMap(t *testing.T) {
+	ctx := context.TODO()
+	name := "test-name"
+	namespace := "eksa-packages"
+	cmdata := make(map[string]string)
+
+	t.Run("golden path for UpdateConfigMap adding one namespace", func(t *testing.T) {
+		add := true
+		cmdata[namespace] = "a"
+		mockClientset := fake.NewSimpleClientset(&v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      varConfigMapName,
+				Namespace: varPackagesNamespace,
+			},
+			Data: cmdata,
+		})
+		ecrAuth := ecrSecret{clientset: mockClientset}
+
+		err := ecrAuth.UpdateConfigMap(ctx, name, namespace, add)
+
+		assert.Nil(t, err)
```

Comment:
```suggestion
		err := ecrAuth.UpdateConfigMap(ctx, name, namespace, add)
		require.NoError(t, err)
```
NoError because it's clearer what you expect. "require", because it will stop the test if the condition doesn't hold. If you have an error there, the next call will almost certainly fail as well, so why litter the test output with more than necessary.

Reply:
done